java/vector/src/main/java/org/apache/arrow/vector/compare/RangeEqualsVisitor.java

```diff
@@ -214,6 +215,18 @@ public Boolean visit(NullVector left, Range range) {
     return true;
   }
 
+  @Override
+  public Boolean visit(ExtensionTypeVector<?> left, Range range) {
+    if (!validate(left) || !(right instanceof ExtensionTypeVector<?>)) {
```

Comment:
For better performance, `!(right instanceof ExtensionTypeVector<?>)` could be placed first?